Description: Don't crash in xsmp_stop on clients which have an ICE connection but not registered yet
Bug: https://bugzilla.gnome.org/show_bug.cgi?id=598211
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/gnome-session/+bug/437425
Origin: https://bugzilla.gnome.org/attachment.cgi?id=145401

Index: gnome-session-2.30.0/gnome-session/gsm-xsmp-client.c
===================================================================
--- gnome-session-2.30.0.orig/gnome-session/gsm-xsmp-client.c	2010-06-04 14:02:38.024237334 +1000
+++ gnome-session-2.30.0/gnome-session/gsm-xsmp-client.c	2010-06-04 14:02:39.494233661 +1000
@@ -725,6 +725,14 @@
                 return FALSE;
         }
 
+        if (xsmp->priv->conn == NULL) {
+                g_set_error (error,
+                             GSM_CLIENT_ERROR,
+                             GSM_CLIENT_ERROR_NOT_REGISTERED,
+                             "Client is not registered");
+                return FALSE;
+        }
+
         SmsDie (xsmp->priv->conn);
 
         return TRUE;
